### PR TITLE
Check that openscap rpms are related to the system arch

### PR DESCRIPTION
**Pull request by pirat89**
OpenSCAP related rpms are built for 64b and 32b arch as well. The
issue occurs on 64b systems where even 32b verions of rpms could
be installed. In case that only 32b rpms are installed on 64b systems
the assessment of the system fails as all modules are skipped with
'notchecked' status.

Related: OAMG-849

### CODE REVIEW

**Comment by bocekm** [comment]
Reproducer:
- Install the `openscap-engine-sce`, `openscap-utils` and `openscap` in both the x86_64 and i686
- Run the `preupg` and observe that all the modules are being skipped

**Comment by bocekm** [comment]
Oh, I missed the part that only i686 of the packages needs to be installed. Since removing the x86_64 versions using yum tries to remove the PA as well, I needed to remove them by `rpm -e openscap.x86_64 openscap-engine-sce.x86_64 --nodeps`.

However, that caused a different error than the one described:
```
001/139 ...running (Configuration files to be reviewed)/usr/bin/oscap: error while loading shared libraries: libopenscap.so.8: cannot open shared object file: No such file or directory

The assessment finished (time 00:00s)
Traceback (most recent call last):
  File "/usr/bin/preupg", line 20, in main
    return_code = app.run()
  File "/usr/lib/python2.6/site-packages/preupg/application.py", line 752, in run
    retval = self.scan_system()
  File "/usr/lib/python2.6/site-packages/preupg/application.py", line 509, in scan_system
    self.prepare_xml_for_html()
  File "/usr/lib/python2.6/site-packages/preupg/application.py", line 344, in prepare_xml_for_html
    self.report_parser.reload_xml(self.openscap_helper.get_default_xml_result_path())
  File "/usr/lib/python2.6/site-packages/preupg/report_parser.py", line 105, in reload_xml
    content = FileHelper.get_file_content(self.path, 'rb', False, False)
  File "/usr/lib/python2.6/site-packages/preupg/utils.py", line 185, in get_file_content
    f = open(full_path, perms)
IOError: [Errno 2] No such file or directory: u'/root/preupgrade/result.xml'
```

What am I doing wrong?

File: preupg/application.py [hunk 1]
```diff
@@ -646,6 +646,41 @@ def determine_module_set_copy_location(self):
...
+        """Check whether expected openscap rpms are installed."""
+        class GetCmdStdout():
+            def __init__(self):
+                self.stdout_lines = None
+
+            def __call__(self, line):
+                if line.strip():
+                    self.stdout_lines.append(line.strip())
+
+        if not os.path.exists(settings.openscap_binary):
```
[the remark this answers is not in the document]
Comment (pirat89) [reply preupg/application.py:660]: I will rather keep it there. Just for sure.

File: preupg/application.py [hunk 2]
```diff
@@ -646,6 +646,41 @@ def determine_module_set_copy_location(self):
...
+            log_message("Oscap with SCE enabled is not installed")
+            return False
+        if not os.access(settings.openscap_binary, os.X_OK):
+            log_message("Oscap with SCE %s is not executable"
+                        % settings.openscap_binary)
+            return False
+        # that's generic problem that could on various rpm-based systems
```
[the remark this answers is not in the document]
Comment (pirat89) [reply preupg/application.py:667]: fixed

**Comment by pirat89** [comment]
@bocekm aaah. I know already what's the difference!! It's because the brew build server and copr has a little bit different configuration. The copr server specifies the `%{_isa}` macro even for noarch (which should not) and in case of brew it is unspecified. So in case of our copr builds, you really have to use the `rpm` ignoring dependencies. Sorry for confusion. I completely forgot about that difference. See the output below

```bash
# copr build
[root@rhel6-upgrade pkgs]$ rpm -q --requires -p preupgrade-assistant-2.6.1-0.202007211433Z.PR366.el6_10.noarch.rpm | grep openscap
warning: preupgrade-assistant-2.6.1-0.202007211433Z.PR366.el6_10.noarch.rpm: Header V3 RSA/SHA1 Signature, key ID f5168da0: NOKEY
openscap(x86-64) >= 0:1.0.8
openscap-engine-sce(x86-64) >= 0:1.0.8
openscap-utils(x86-64) >= 0:1.0.8
[root@rhel6-upgrade pkgs]$
# brew build 
[root@rhel6-upgrade pkgs]$ rpm -q --requires -p preupgrade-assistant-2.6.1-1.el6.noarch.rpm | grep openscap
openscap >= 0:1.0.8
openscap-engine-sce >= 0:1.0.8
openscap-utils >= 0:1.0.8
```

**Comment by pirat89** [comment]
# Results of testing:
## broken set of openscap rpms (openscap-engine-sce for x86_64 is missing) for PA (print the msg and stop)
```bash
# rpm -qa | grep openscap
openscap-1.2.13-2.el6.i686
openscap-utils-1.2.13-2.el6.x86_64
openscap-1.2.13-2.el6.x86_64
openscap-engine-sce-1.2.13-2.el6.i686
openscap-scanner-1.2.13-2.el6.x86_64
[root@rhel6-upgrade pkgs]# preupg
The Preupgrade Assistant is a diagnostics tool 
and does not perform the actual upgrade.
Do you want to continue? [Y/n]
y
The openscap-engine-sce rpm is not installed for the x86_64 architecture. This usually ends in a broken state in which all the Preupgrade Assistant modules are skipped (notchecked state). Please, install packages related for your architecture. See https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html-single/6.10_release_notes/index#BZ1804691 for more info.
```

## correct set of rpms installed (all required rpms for specific arch are installed)
```bash
# preupgs
Assessment of the system, running checks / SCE scripts:
001/140 ...done    (Configuration files to be reviewed) (time: 00:01s)
002/140 ...done    (File lists for the manual migration) (time: 00:00s)
....
|Configuration files to be reviewed                                                            |needs_action      |
|MySQL configuration                                                                           |needs_action      |
|Packages not signed by Red Hat                                                                |needs_action      |
|Removed RPM packages                                                                          |needs_action      |
|"not-base" channels                                                                           |needs_action      |
...
```

**Pull request merged**